### PR TITLE
Events Pagination, data model, update endpoint

### PR DESCRIPTION
**Pull request by gumityolcu**
Pagination is done via 'mongoose-paginate' package.

GET '/api/events' endpoint needs 2 URL parameters: skip and limit. GET '/api/events/getEventbyOwner' needs 3 parameters: id, skip and limit.

In both cases, if skip is omitted, it is assumed 0, if limit is omitted it is assumed 10. If id is omitted an error message is sent back.

-------------------------------

Update:
Event update endpoint is added.
Event data model is updated to include artists field.

---------------------------------

Update 2:
I am now starting to wonder if putting these functionalities in one branch was a good idea.

Anyways, I noticed the code wasn't working and edited the code. Event update endpoint allows to update blockedUsers, name, price, date, artists and description fields. Since a user shouldn't be able to change who will attend or not attend the event but she can block users from an event. Also, the user field holds the owner id of the event therefore is not updatable. This is not consistent with our current front end, about which i will open an issue.

I think we should merge this branch because it is getting complex. I will not work on this branch apart from fixing issues relating to these functionalities.

### CODE REVIEW

File: backend/controllers/EventController.js [hunk 1]
```diff
@@ -2,6 +2,10 @@ var mongoose = require("mongoose");
 var Event = require("../models/Event");
 
 
+
```
Comment (okyksl) [inline backend/controllers/EventController.js]: Can we remove extra spaces @gumityolcu?

File: backend/npm-debug.log [hunk 2]
```diff
@@ -0,0 +1,45 @@
+0 info it worked if it ends with ok
```
Comment (okyksl) [inline backend/npm-debug.log]: I think this file shouldn't be pushed to the repo.

**Comment by gumityolcu** [comment]
Yes, it is pushed due to inattentive usage of git commit -a

File: backend/npm-debug.log (hunk 2 above)
Comment (gumityolcu) [reply backend/npm-debug.log]: This is handled.

File: backend/controllers/EventController.js (hunk 1 above)
Comment (gumityolcu) [reply backend/controllers/EventController.js]: Done.

File: backend/controllers/EventController.js [hunk 3]
```diff
@@ -44,33 +48,77 @@ exports.getEventbyId = function(req, res, next)
...
 }
+
+
+// needs params: id(owner id as string), skip(integer, default 0), limit(integer, default 10)
+// will return array of event objects with <limit> elements starting from object number <skip> in the db 
 exports.getEventbyOwner = function(req,res,next)
 {
-  console.log("Searching owner : " + req.params.id);
-    Event.find({'owner' : req.params.id}, function(err, docs)
+  console.log("Searching owner : " + req.query.id);
```
[the remark this answers is not in the document]
Comment (gumityolcu) [reply backend/controllers/EventController.js]: This is handled.

**Comment by gumityolcu** [comment]
I have handled the points in your review, also deleted the console statements Codacy doesn't stop complaining about :smile:

**Comment by gumityolcu** [comment]
I resolved the conflicts after change in master.

**Comment by serkan-ozel-2015400123** [comment]
:) what is this pr have suffered, what is problem here

**Comment by okyksl** [comment]
@nazmicancalik @byklyci @akaogluozgur could you please checkout this PR?

**Comment by gumityolcu** [comment]
I added minor changes about event data model and event update endpoint, since they are not features of major scope. I am changing the name of the PR accordingly, please don't forget to review these too.

**Comment by okyksl** [comment]
Does this PR closes #219, #229 @gumityolcu?

**Comment by gumityolcu** [comment]
@okyksl Yes, this PR closes #219, #229 and also it has closed #203 previously.

All these issues piled up in one PR because of amateur usage of versioning system.

**Pull request merged**